Assignment2/Logger/BaseLoggerMixins.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+
+namespace Logger
+{
+    public static class BaseLoggerMixins
+    {
+        public static void Error(this BaseLogger baseLogger, string message, params object[] messageArgs)
+        {
+            if (baseLogger is null)
+            {
+                throw new ArgumentException("The BaseLogger can not be null.");
```

Comment:
Since `BaseLogger` is an abstract class, this really doesn't give the user the information it could. Instead, like the prompt says, we should use `nameof` here to get the name of the child logger class.
```suggestion
                throw new ArgumentException($"The {nameof(logger)} can not be null.");
```